packages/channels/src/slack/stream.ts

```diff
@@ -0,0 +1,26 @@
+import { ChannelContext } from '../base/context'
+import { CardToCarouselRenderer } from '../base/renderers/card'
+import { ChannelSender } from '../base/sender'
+import { TypingSender } from '../base/senders/typing'
+import { ChannelStream } from '../base/stream'
+import { SlackContext } from './context'
+import { SlackRenderers } from './renderers'
+import { SlackSenders } from './senders'
+import { SlackService } from './service'
+
+export class SlackStream extends ChannelStream<SlackService, SlackContext> {
+  get renderers() {
+    return [new CardToCarouselRenderer(), ...SlackRenderers]
+  }
+
+  get senders(): ChannelSender<ChannelContext<any>>[] {
+    return [new TypingSender(), ...SlackSenders]
+  }
+
+  protected async getContext(base: ChannelContext<any>): Promise<SlackContext> {
+    return {
+      ...base,
+      message: { blocks: [] } as any
```

Comment:
as any? Also, since we init blocks with an empty array. Which I think will always be the default, couldn't we change its typings so it's not optional anymore? This would allow us to remove a lot of null checking ('?').

Reply:
Good point. Removed the as any and added typings to avoid null checking